packages/vite/src/node/config.ts

```diff
@@ -477,12 +477,34 @@ export async function resolveConfig(
   const resolvedRoot = normalizePath(
     config.root ? path.resolve(config.root) : process.cwd(),
   )
+
+  // We will warn if the root path includes either the '#' or '?' character
+  const badPathCharacters = []
+
   if (resolvedRoot.includes('#')) {
+    badPathCharacters.push('#')
+  }
+
+  if (resolvedRoot.includes('?')) {
+    badPathCharacters.push('?')
+  }
+
+  if (badPathCharacters.length > 0) {
+    const characterString =
+      badPathCharacters.length > 1
+        ? badPathCharacters.map((c) => `"${c}"`).join(' and ')
+        : badPathCharacters.map((c) => `"${c}"`).join('')
+
+    const inflectedChars =
+      badPathCharacters.length > 1 ? 'characters' : 'character'
+
+    // FIXME: Shouldn't this fail hard? There's no way paths with either
+    //        # or ? in them will be resolved correctly?
```

Comment:
I don't think we want to fail hard for it for now. We can revisit this in a major perhaps if this is something we really don't want to support, and we can error there in the future.

Reply:
👍